marshmallow/utils.py

```diff
@@ -67,7 +67,7 @@ def is_indexable_but_not_string(obj):
 
 def is_collection(obj):
     """Return True if ``obj`` is a collection type, e.g list, tuple, queryset."""
-    return is_iterable_but_not_string(obj) and not isinstance(obj, Mapping)
+    return is_iterable_but_not_string(obj) and not isinstance(obj, Collection)
```

Comment:
This logic was saying "is collection if iterable but not a string and not a mapping". Now it says "is collection if iterable but not a string and not a collection", which is not right. I think you might was something more like "is collection if a collection but not a string".

Reply:
The reason it is done this way is probably, because there is no `Collection` abstract class in python 2. Starting with a more generic base class and excluding the bits that aren't collections should be a pretty inclusive strategy.

Reply:
Thanks @deckar01, I somehow completely misread this. In that case, I'll cancel this PR in favor of whatever changes you think should go in to improve interop (e.g. with werkzeug Headers) if any. At the very least maybe add a comment in the code that Python 2's lack of collections.Collection is a reason this exists?

Reply:
I'm not familiar with the werkzeug headers problem. Can you create an issue and provide an example that reproduces the bug?